com.unity.ml-agents/Runtime/Inference/BarracudaModelParamLoader.cs

```diff
@@ -138,7 +138,7 @@ SensorComponent[] sensorComponents
             for (var sensorIndex = 0; sensorIndex < sensorComponents.Length; sensorIndex++)
             {
                 var sensor = sensorComponents[sensorIndex];
-                if (!sensor.IsVisual())
+                if (sensor.GetObservationShape().Length != 3)
```

Comment:
Don't we need a similar check for dim==2 observations?

Reply:
Note that this block is just checking that the tensor is there at all, whereas the actual shape compatibility is in CheckRankTwoObsShape.